crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -254,6 +254,19 @@ def prepare(options = {})
 
       protected
 
+      def finalize_cluster_upgrade(cluster)
+        save_upgrade_state("Finalizing the upgrade of nodes in cluster #{cluster}")
+
+        NodeObject.find("pacemaker_config_environment:#{cluster}").each do |node|
+          if node["pacemaker"]["founder"] && node.roles.include?("neutron-network")
+            node.run_ssh_cmd("source /root/.openrc; neutron-ha-tool --l3-agent-rebalance")
+          end
+          node["crowbar_wall"]["crowbar_upgrade_step"] = "done"
+          node.save
+        end
+        # FIXME: we need to enable chef-client and crowbar-join services at some point
```

Comment:
Enabling `crowbar-join` and `chef-client` directly from `node_api.join_and_chef` would work and not break sync marks, but it would be reset by the same chef-client run when executing `crowbar_upgrade` recipe (https://github.com/crowbar/crowbar-core/blob/master/chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb#L67).

So I think we need another `crowbar_upgrade_step`, that will 1. tell crowbar_upgrade to start the service again and 2. not break skipping of pacemaker sync marks (as that is needed until we can declare whole cluster ready)

Maybe it's overcomplicated, but I do not see other way now...